OptimoveSDK/optimove-sdk/src/main/java/com/optimove/sdk/optimove_sdk/optimobile/OptimobileConfig.java

```diff
@@ -0,0 +1,366 @@
+        this.inAppConsentStrategy = strategy;
+    }
+
+    private void setCname(@Nullable URL deepLinkCname) {
+        this.deepLinkCname = deepLinkCname;
+    }
+
+    private void setDeferredDeepLinkHandler(DeferredDeepLinkHandlerInterface deferredHandler) {
+        this.deferredDeepLinkHandler = deferredHandler;
+    }
+
+    private void setMinLogLevel(@Nullable LogLevel minLogLevel){
+        this.minLogLevel = minLogLevel;
+    }
+
+    @Nullable
+    String getApiKey() {
+        return apiKey;
+    }
+
+    @Nullable
+    String getSecretKey() {
+        return secretKey;
+    }
+
+    public @Nullable String getOptimoveToken() {
+        return optimoveToken;
+    }
+
+    public @Nullable String getConfigFileName() {
+        return configFileName;
+    }
+
+    public @DrawableRes
+    int getNotificationSmallIconId() {
+        return notificationSmallIconId;
+    }
+
+    public int getSessionIdleTimeoutSeconds() {
+        return sessionIdleTimeoutSeconds;
+    }
+
+    JSONObject getRuntimeInfo() {
+        return this.runtimeInfo;
+    }
+
+    JSONObject getSdkInfo() {
+        return this.sdkInfo;
+    }
+
+    @Nullable Map<UrlBuilder.Service, String> getBaseUrlMap() {
+        return baseUrlMap;
+    }
+
+    InAppConsentStrategy getInAppConsentStrategy() {
+        return inAppConsentStrategy;
+    }
+
+    public @Nullable
+    URL getDeepLinkCname() {
+        return this.deepLinkCname;
+    }
+
+    public DeferredDeepLinkHandlerInterface getDeferredDeepLinkHandler() {
+        return this.deferredDeepLinkHandler;
+    }
+
+    public boolean isOptimoveConfigured(){
+        return this.optimoveToken != null && this.configFileName != null ;
+    }
+
+    public boolean isOptimobileConfigured(){
+        return this.apiKey != null && this.secretKey != null;
+    }
+
+    public @Nullable LogLevel getCustomMinLogLevel(){
+        return this.minLogLevel;
+    }
+
+    /**
+     * Config builder for the Optimobile client
+     */
+    public static class Builder {
+        private @Nullable
+        String apiKey;
+        private @Nullable
+        String secretKey;
+        private @Nullable
+        String optimoveToken;
+        private @Nullable
+        String configFileName;
+
+        @DrawableRes
+        private int notificationSmallIconDrawableId = OptimobileConfig.DEFAULT_NOTIFICATION_ICON_ID;
+        private InAppConsentStrategy consentStrategy = null;
+        private int sessionIdleTimeoutSeconds = OptimobileConfig.DEFAULT_SESSION_IDLE_TIMEOUT_SECONDS;
+
+        private JSONObject runtimeInfo;
+        private JSONObject sdkInfo;
+        private @Nullable Map<UrlBuilder.Service, String> baseUrlMap;
+
+        private @Nullable
+        URL deepLinkCname;
+        private DeferredDeepLinkHandlerInterface deferredDeepLinkHandler;
+
+        private @Nullable LogLevel minLogLevel;
+
+        public Builder(@Nullable String optimoveCredentials, @Nullable String optimobileCredentials) {
```

Comment:
Can we call these "optimoveCredentials" and "mobileCredentials" to line up with the places in the portal where they'll copy it from.

Reply:
mb we need to line up portal? in sdk Kumulos === optimobile --> optimobileCredentials